### PR TITLE
introduces optional IPNFT symbols

### PR DESCRIPTION
**Pull request by elmariachi111**
- overrides the mint function instead of replacing it 
- IPNFT owners can update the symbol at will
- emits `SymbolUpdated` events
- picked up by subgraph
- thoroughly tested that subgraphs update old and new mints correctly
- deployment scripts that help testing V21 -> V22 upgrades

https://www.notion.so/moleculeto/Add-an-individual-onchain-prefix-symbol-to-each-IP-NFT-3e7c609e4d444f6f8659010cdf4232d6

image

### CODE REVIEW

**Comment by elmariachi111** [comment]
just when looking at it, I don't like the overrides here. Will change it again, stdby

File: src/old/IPNFTV21.sol [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  | ▓▓ | ▓▓▓▓▓▓▓ \▓▓▓▓▓▓ ▓▓\▓▓ ▓▓ ▓▓▓▓▓     | ▓▓
+ _| ▓▓_| ▓▓            | ▓▓ \▓▓▓▓ ▓▓        | ▓▓
+|   ▓▓ \ ▓▓            | ▓▓  \▓▓▓ ▓▓        | ▓▓
+ \▓▓▓▓▓▓\▓▓             \▓▓   \▓▓\▓▓         \▓▓
+ */
+
+/// @title IPNFT V2.1
+/// @author molecule.to
+/// @notice IP-NFTs capture intellectual property to be traded and fractionalized
+contract IPNFTV21 is
```
Comment (firattale) [inline src/old/IPNFTV21.sol:38]: I wonder, why do we keep the old version ?

Comment (elmariachi111) [reply src/old/IPNFTV21.sol:38]: to test upgrades from the old to the new one :)

File: src/IPNFT.sol [hunk 2]
```diff
@@ -134,15 +138,29 @@ contract IPNFT is
         return reservationId;
     }
 
+    /**
+     * @notice deprecated: the old interface without a symbol.
+     */
+    function mintReservation(address to, uint256 reservationId, uint256 mintPassId, string memory tokenURI)
```
Comment (firattale) [inline src/IPNFT.sol:144]: Are we going to keep this func? As far as I see we are sending empty string on FE if the user doesn't give an input, so seems like this first mintReservation imp. will be obsolete :)

Comment (elmariachi111) [reply src/IPNFT.sol:144]: but it's still usable. Imo a symbol shouldn't be *mandatory* to mint an IPNFT.

**Review by firattale — APPROVED** [review]
We tested on a local build, and it worked fine, lgtm!
Screen Shot 2023-03-17 at 10 10 18

**Comment by firattale** [comment]
I added this line `POSTGRES_INITDB_ARGS: "-E UTF8 --locale=C"` to docker file to use latest graph-node image.

https://github.com/graphprotocol/graph-node/pull/4151/files

**Pull request merged**